test/functional/client_side_encryption_corpus_tests.js

```diff
@@ -0,0 +1,381 @@
+'use strict';
+
+// The corpus test exhaustively enumerates all ways to encrypt all BSON value types. Note, the test data includes BSON binary subtype 4 (or standard UUID), which MUST be decoded and encoded as subtype 4. Run the test as follows.
+
+const fs = require('fs');
+const path = require('path');
+const EJSON = require('mongodb-extjson');
+const chai = require('chai');
+const expect = chai.expect;
+chai.config.includeStack = true;
+chai.config.showDiff = true;
+chai.config.truncateThreshold = 0;
+
+describe('Client Side Encryption Corpus', function() {
+  // See if we can run these tests
+  if (process.env.AWS_ACCESS_KEY_ID == null || process.env.AWS_SECRET_ACCESS_KEY == null) {
+    console.log('skipping Client Side Encryption Corpus tests due to lack of AWS credentials');
+    return;
+  }
+
+  let mongodbClientEncryption;
+  try {
+    mongodbClientEncryption = require('mongodb-client-encryption')(require('../../index'));
+  } catch (e) {
+    console.log(
+      'skipping Client Side Encryption Corpus tests due to inability to load mongodb-client-encryption'
+    );
+    return;
+  }
+
+  const corpusDir = path.resolve(__dirname, 'spec', 'client-side-encryption', 'corpus');
+  function loadCorpusData(filename) {
+    return EJSON.parse(fs.readFileSync(path.resolve(corpusDir, filename), { strict: true }));
+  }
+
+  // TODO: build this into EJSON
+  // TODO: make a custom chai assertion for this
+  function toJSONComp(value) {
```

Comment:
nit: can this have a better name? I think its a function that converts a value into something we can use for a strict JSON object comparison?

Reply:
Also, isn't this what [EJSON.serialize](https://github.com/mongodb-js/mongodb-extjson/blob/master/lib/ext_json.js#L194-L197) (in the newer bson library as well [here](https://github.com/mongodb/js-bson/blob/master/lib/extended_json.js#L200)) does?

Reply:
But the first link I put there is to the existing `mongodb-extjson` library, which you are using here. I think you have access to the method.

Reply:
that's in `v3`, which requires the later versions of bson :(.

Reply:
oof, okay